File: app/src/main/java/org/eclipsesoundscapes/data/DataManager.java
package org.eclipsesoundscapes.data;

import androidx.annotation.Nullable;
import androidx.core.util.Pair;

public class DataManager {

    private SharedPrefsHelper sharedPrefsHelper;

    private Pair<Integer, Integer> featuresPosition;

    public DataManager(SharedPrefsHelper sharedPrefsHelper) {
        this.sharedPrefsHelper = sharedPrefsHelper;
    }

    public void setFirstContact(String date){
        sharedPrefsHelper.saveFirstContact(date);
    }

    public void setTotality(String date){
        sharedPrefsHelper.saveTotality(date);
    }

    public void setSimulated(boolean simulated){
        sharedPrefsHelper.saveSimulated(simulated);
    }

    public void setLocationAccess(boolean access){
        sharedPrefsHelper.saveLocationAccess(access);
    }

    public void setRequestedLocation(boolean requested){
        sharedPrefsHelper.saveRequestedLocation(requested);
    }

    public void setNotification(boolean canSendNotifications){
        sharedPrefsHelper.saveNotification(canSendNotifications);
    }

    public void setWalkthroughComplete(boolean completed){
        sharedPrefsHelper.saveWalkthroughComplete(completed);
    }

    public void setLanguage(final String language) {
        sharedPrefsHelper.setPreferredLanguage(language);
    }

    public String getLanguage() {
        return sharedPrefsHelper.getPreferredLanguage();
    }

    public void setRumblingCheckpoint(String eclipseId, String coordinates){
        sharedPrefsHelper.setCheckpoint(eclipseId, coordinates);
    }

    public String getFirstContact(){
        return sharedPrefsHelper.getFirstContact();
    }

    public String getTotality(){
        return sharedPrefsHelper.getTotality();
    }

    public String getRumblingCheckpoint(String eclipseId){
        return sharedPrefsHelper.getCheckpoint(eclipseId);
    }

    public boolean getSimulated(){
        return sharedPrefsHelper.getSimulated();
    }

    public boolean getWalkthroughComplete(){
        return sharedPrefsHelper.getWalkthroughComplete();
    }

    public boolean getLocationAccess(){
        return sharedPrefsHelper.getLocationAccess();
    }

    public boolean getRequestedLocation(){
        return sharedPrefsHelper.getRequestedLocation();
    }

    public boolean getNotifications(){
        return sharedPrefsHelper.getNotifications();
    }

    public void saveFeaturesPosition(int page, int tab) {
        featuresPosition = new Pair(page, tab);
    }

    @Nullable
    public Pair<Integer, Integer> getFeaturesPosition() {
        return featuresPosition;
    }
}

File: app/src/main/java/org/eclipsesoundscapes/data/DataManager.kt
package org.eclipsesoundscapes.data

import android.location.Location
import androidx.core.util.Pair

class DataManager(private val sharedPrefsHelper: SharedPrefsHelper) {
    var featuresPosition: Pair<Int, Int>? = null
        private set

    fun saveFeaturesPosition(page: Int, tab: Int) {
        featuresPosition = Pair<Int, Int>(page, tab)
    }

    var language: String?
        get() = sharedPrefsHelper.preferredLanguage
        set(language) {
            sharedPrefsHelper.preferredLanguage = language
        }

    var firstContact: String?
        get() = sharedPrefsHelper.firstContact
        set(date) {
            sharedPrefsHelper.saveFirstContact(date)
        }

    var totality: String?
        get() = sharedPrefsHelper.totality
        set(date) {
            sharedPrefsHelper.saveTotality(date)
        }

    var simulated: Boolean
        get() = sharedPrefsHelper.simulated
        set(simulated) {
            sharedPrefsHelper.saveSimulated(simulated)
        }

    var walkthroughComplete: Boolean
        get() = sharedPrefsHelper.walkthroughComplete
        set(completed) {
            sharedPrefsHelper.saveWalkthroughComplete(completed)
        }

    var locationAccess: Boolean
        get() = sharedPrefsHelper.locationAccess
        set(access) {
            sharedPrefsHelper.saveLocationAccess(access)
        }

    var requestedLocation: Boolean
        get() = sharedPrefsHelper.requestedLocation
        set(requested) {
            sharedPrefsHelper.saveRequestedLocation(requested)
        }

    var skippedLocationsPermission: Boolean
        get() = sharedPrefsHelper.skippedLocationPermission
        set(requested) {
            sharedPrefsHelper.saveSkippedLocationPermission(requested)
        }

    var requestedNotification: Boolean
        get() = sharedPrefsHelper.requestedNotifications
        set(requested) {
            sharedPrefsHelper.saveRequestedNotifications(requested)
        }

    var skippedNotificationsPermission: Boolean
        get() = sharedPrefsHelper.skippedNotificationsPermission
        set(requested) {
            sharedPrefsHelper.saveSkippedNotificationsPermission(requested)
        }

    val notifications: Boolean
        get() = sharedPrefsHelper.notifications

    var currentEclipseDate: String?
        get() = sharedPrefsHelper.eclipseDate
        set(eclipseDate) {
            sharedPrefsHelper.eclipseDate = eclipseDate
        }

    var lastLocation: Location?
        get() = sharedPrefsHelper.lastLocation
        set(location) {
            sharedPrefsHelper.lastLocation = location
        }

    fun setRumblingCheckpoint(eclipseId: String?, coordinates: String?) {
        sharedPrefsHelper.setCheckpoint(eclipseId, coordinates)
    }

    fun getRumblingCheckpoint(eclipseId: String?): String {
        return sharedPrefsHelper.getCheckpoint(eclipseId)
    }

    fun setNotification(canSendNotifications: Boolean) {
        sharedPrefsHelper.saveNotification(canSendNotifications)
    }
}